docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +79,36 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an object option.
+
+When `reservedFirst` is defined, React reserved props (`children`, `dangerouslySetInnerHTML`, `key`, `ref`) must be listed before all other props, but still respecting the alphabetical order:
+
+```jsx
+<Hello key={0} ref="John" name="John" />
+```
+
+If given as an object, the option accepts two optional sub-options:
+
+#### `reservedList`
+
+Use this to optionally override the default list of reserved props. **Note**: if given the values in the list may only be a **subset** of React reserved props.
+
+With `reservedList: ["key"]`, the following will not warn:
+
+```jsx
+<Hello key={'uuid'} name="John" ref="ref" />
+```
+
+#### `noSortReservedAlphabetically`
+
+When `true`, alphabetical sorting of reserved props is not enforced, but alphabetical sorting of other props still is, unless `noSortAlphabetically` is set to `true` as well.
```

Comment:
Oh, I was just trying to match the other options, wasn't trying to create new functionality. I'll update so that setting `noSortAlphabetically` overrides `noSortReservedAlphabetically`